src/test/java/org/swisspush/redisques/QueueStatsServiceTest.java

```diff
@@ -0,0 +1,191 @@
+    }
+
+    @Test
+    public void testDequeueStatsNotCalledWhenIntervalIsZeroOrNegative(TestContext testContext) {
+        // Mock the config so that getDequeueStatisticReportIntervalSec returns 0 or a negative value
+        Mockito.when(config.getDequeueStatisticReportIntervalSec()).thenReturn(0); // Value equal to 0
+        // Alternatively: Mockito.when(config.getDequeueStatisticReportIntervalSec()).thenReturn(-1); // For negative value
+
+        // Set up async for the test
+        Async async = testContext.async();
+
+        // Mock a context and mentor
+        Object mockContext = new Object();
+        QueueStatsService.GetQueueStatsMentor<Object> mentor = new QueueStatsService.GetQueueStatsMentor<>() {
+            @Override
+            public boolean includeEmptyQueues(Object ctx) {
+                return true;
+            }
+
+            @Override
+            public int limit(Object ctx) {
+                return 10;
+            }
+
+            @Override
+            public String filter(Object ctx) {
+                return "*";
+            }
+
+            @Override
+            public void onQueueStatistics(java.util.List<QueueStatsService.Queue> queues, Object ctx) {
+                // Assert the dequeue stats were not attached
+                testContext.assertTrue(queues.get(0).getNextDequeueDueTimestampEpochMs() == null);
+                async.complete();
+            }
+
+            @Override
+            public void onError(Throwable ex, Object ctx) {
+                testContext.fail(ex);
+            }
+        };
+
+        // Mock fetchQueueNamesAndSize to return a valid list of queues
+        Mockito.doAnswer(invocation -> {
+            BiConsumer<Throwable, QueueStatsService.GetQueueStatsRequest<Object>> callback = invocation.getArgument(1);
+            QueueStatsService.GetQueueStatsRequest<Object> request = invocation.getArgument(0);
+            request.queues = Collections.singletonList(new QueueStatsService.Queue("testQueue", 1));
+            callback.accept(null, request);
+            return null;
+        }).when(queueStatsService).fetchQueueNamesAndSize(Mockito.any(), Mockito.any());
+
+        // Mock fetchRetryDetails to do nothing and call the next step
+        Mockito.doAnswer(invocation -> {
+            BiConsumer<Throwable, QueueStatsService.GetQueueStatsRequest<Object>> callback = invocation.getArgument(1);
+            QueueStatsService.GetQueueStatsRequest<Object> request = invocation.getArgument(0);
+            callback.accept(null, request);
+            return null;
+        }).when(queueStatsService).fetchRetryDetails(Mockito.any(), Mockito.any());
+
+        // Call getQueueStats to trigger the flow
+        queueStatsService.getQueueStats(mockContext, mentor);
+
+        // Verify that attachDequeueStats was NOT called
+        Mockito.verify(queueStatsService, Mockito.never()).attachDequeueStats(Mockito.any(), Mockito.any());
+
```

Comment:
The problem in the `QueueStatsService` class should be visible here. Do not only test for `attachDequeueStats()` never be called, but also for the `GetQueueStatsMentor` to have answered the request.